### PR TITLE
Adding mysql integration tests

### PR DESCRIPTION
**Pull request by samiura**
screen

### CODE REVIEW

**Review by atoulme — APPROVED** [review]
LGTM, good to merge if tests pass

File: tests/receivers/mysql/mysql_test.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import (
+	"path"
+	"testing"
+
+	"github.com/signalfx/splunk-otel-collector/tests/testutils"
+)
+
+var mysql = []testutils.Container{testutils.NewContainer().WithContext(
+	path.Join(".", "testdata", "server"),
+).WithName("mysql").WithExposedPorts("3306:3306").WillWaitForPorts("3306").WillWaitForLogs("X Plugin ready for connections. Bind-address:")}
```
Comment (crobert-1) [inline tests/receivers/mysql/mysql_test.go:29]: Nit: `3306` is used a few times in this file, maybe move it to a variable?

File: tests/receivers/mysql/testdata/server/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,10 @@
+FROM mysql:8.2.0
+
+RUN chown -R mysql:root /var/lib/mysql/
+
+ENV MYSQL_ROOT_PASSWORD='testuser'
+
+COPY initMySqlDB.sql /docker-entrypoint-initdb.d/
+
```
Comment (crobert-1) [inline tests/receivers/mysql/testdata/server/Dockerfile:8]:
```suggestion
```

File: tests/receivers/mysql/testdata/all_metrics_config.yaml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    username: "${MYSQLDB_USERNAME}"
+    password: "${MYSQLDB_PASSWORD}"
+    database: otel
+    tls:
+      insecure: true
+    statement_events:                                                                                                                                                                                                           
+      digest_text_limit: 120                                                                                                                                                                                                    
+      time_limit: 24h                                                                                                                                                                                                           
+      limit: 250         
+exporters:
```
Comment (crobert-1) [inline tests/receivers/mysql/testdata/all_metrics_config.yaml:13]:
```suggestion

exporters:
```

File: tests/receivers/mysql/testdata/all_metrics_config.yaml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  otlp:
+    endpoint: "${OTLP_ENDPOINT}"
+    tls:
+      insecure: true
+
```
Comment (crobert-1) [inline tests/receivers/mysql/testdata/all_metrics_config.yaml:18]:
```suggestion
```

**Review by jvoravong — APPROVED** [review]

**Review by crobert-1 — APPROVED** [review]
Just some minor nits, thanks for adding tests!

File: tests/receivers/mysql/testdata/resource_metrics/all.yaml [hunk 5]
```diff
@@ -0,0 +1,760 @@
+resource_metrics:
+  - attributes:
+      mysql.instance.endpoint: <ANY>
+    scope_metrics:
+      - instrumentation_scope:
+          attributes: {}
+          name: otelcol/mysqlreceiver
+          version: <ANY>
```
Comment (rmfitzpatrick) [inline tests/receivers/mysql/testdata/resource_metrics/all.yaml:8]:
```suggestion
          version: <VERSION_FROM_BUILD>
```

File: tests/receivers/mysql/mysql_test.go [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+package tests
+
+import (
+	"path"
+	"testing"
+
+	"github.com/signalfx/splunk-otel-collector/tests/testutils"
+)
+
+var mysql = []testutils.Container{testutils.NewContainer().WithContext(
```
Comment (rmfitzpatrick) [inline tests/receivers/mysql/mysql_test.go:27]: nit, until this is shared no need to make global

File: tests/receivers/mysql/mysql_test.go [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+	path.Join(".", "testdata", "server"),
+).WithName("mysql").WithExposedPorts("3306:3306").WillWaitForPorts("3306").WillWaitForLogs("X Plugin ready for connections. Bind-address:")}
+
+// This test ensures the collector can connect to a MySQL DB, and properly get metrics. It's not intended to
+// test the receiver itself.
```
Comment (rmfitzpatrick) [inline tests/receivers/mysql/mysql_test.go:32]: wading into semantics, these tests are partially intended to test the underlying components. Though some of the expectations we have about the reported telemetry may need to be updated (resource metric file changes), they can detect breaking changes or bugs that may have been missed upstream.

**Pull request merged**